packages/interactivity/src/hooks.tsx

```
@@ -223,8 +225,13 @@ const resolve = ( path: string, namespace: string ) => {
 		context: getScope().context[ namespace ],
 	};
 	try {
-		// TODO: Support lazy/dynamically initialized stores
-		return path.split( '.' ).reduce( ( acc, key ) => acc[ key ], current );
+		return path
+			.split( '.' )
+			.reduce(
+				( acc, key ) =>
+					key in acc ? acc[ key ] : acc[ key ] || NOT_RESOLVED,
```

Ideally, instead of returning `NOT_RESOLVED`, this value could be thrown and captured inside the `Directives` component. That way, directives wouldn't need to handle this value. The same applies to other errors thrown by the `evaluate()` function.

The only drawback I see is when you call `evaluate()` from callbacks, outside the `Directives` component. In that case, the `evaluate()` call should be done inside a `try..catch`. But that's not so problematic; we wouldn't even have to handle that specific case. If `evaluate()` throws, whatever the error is, that means there's nothing to evaluate. 🤔

---

I have updated the implementation to support only getters, so this conversation is no longer relevant.